File: php/src/Types/UserInfoResponse.php
<?php

namespace Authdog\Types;

/**
 * Response from the /v1/userinfo endpoint
 */
class UserInfoResponse
{
    public Meta $meta;
    public Session $session;
    public User $user;

    public function __construct(array $data)
    {
        $this->meta = new Meta($data['meta'] ?? []);
        $this->session = new Session($data['session'] ?? []);
        $this->user = new User($data['user'] ?? []);
    }

    public function getMeta(): Meta
    {
        return $this->meta;
    }

    public function getSession(): Session
    {
        return $this->session;
    }

    public function getUser(): User
    {
        return $this->user;
    }
}

File: csharp/Types/UserInfoResponse.cs
using Newtonsoft.Json;

namespace Authdog.Types
{
    /// <summary>
    /// Response from the /v1/userinfo endpoint
    /// </summary>
    public class UserInfoResponse
    {
        [JsonProperty("meta")]
        public Meta Meta { get; set; } = new();

        [JsonProperty("session")]
        public Session Session { get; set; } = new();

        [JsonProperty("user")]
        public User User { get; set; } = new();
    }
}
